lib/filter.js

```diff
@@ -0,0 +1,86 @@
+'use strict';
+
+//////////////////////////////
+// contains, is, isNot, doesNotContain, exists, matches
+//
+// Include
+//////////////////////////////
+const addModule = function (results, pkg) {
+  try {
+    results[pkg.package.name] = require(pkg.package.name);
+  }
+  catch (e) {
+    if (!results.hasOwnProperty('__unrequireable')) {
+      results.__unrequireable = [];
+    }
+    results.__unrequireable.push(pkg.package.name);
+  }
+  return results;
+}
+
+module.exports = function ModuleFilter(packages, key) {
+  /**
+    * Contains
+  **/
+  this.containsSync = (search) => {
+    var results = {};
```

Comment:
@Snugug the map function could shorten these sections.

ex:
```
  this.containsSync = (search) => {
    return packages.map(pkg => {
      if (pkg.package[key] && pkg.package[key].indexOf(search) >= 0) {
        return addModule(results, pkg);
      }
    })
  };
```

map is chain-able too, among other things...

ref:

* https://ryanpcmcquen.org/javascript/2015/10/25/map-vs-foreach-vs-for.html
* https://stackoverflow.com/questions/3034392/what-use-does-the-javascript-foreach-method-have-that-map-cant-do/4927981#4927981